pkg/clusterapi/identity.go

```diff
@@ -103,19 +103,36 @@ func configureOIDCInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane,
 	}
 }
 
+func configureAPIServerExtraArgsInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, apiServerExtraArgs map[string]string) {
+	if apiServerExtraArgs == nil {
+		return
+	}
+
+	for k, v := range apiServerExtraArgs {
+		kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.ExtraArgs[k] = v
+	}
+}
+
 func configurePodIamAuthInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, podIamConfig *v1alpha1.PodIAMConfig) {
 	if podIamConfig == nil {
 		return
 	}
 
 	apiServerExtraArgs := kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.ExtraArgs
 	for k, v := range PodIAMAuthExtraArgs(podIamConfig) {
+		if k == "service-account-issuer" {
+			if _, ok := apiServerExtraArgs[k]; ok {
+				apiServerExtraArgs[k] += "," + v
+				continue
```

Comment:
Nit: Instead of continue here, could we just set the string to v initially, and then append to that string in this case so that it will still be assigned through the below line?

Reply:
Updated